src/context.rs

```diff
@@ -4,13 +4,24 @@
 //! internationalization formatters, functions, environmental variables and are expected to be used
 //! together.
 
+use std::cell::RefCell;
 use std::collections::HashMap;
 
 use super::resolve::{Env, ResolveValue};
 use super::syntax::ast;
 use super::syntax::parse;
 use super::types::FluentValue;
 
+enum Variant<'ctx> {
```

Comment:
Can you call it `Entry` please? Variants are used in FTL as a public concept. Entries are in AST but are just a bag of message/comment/junk, so I'm ok having another Entry in Context that is a bad of message/term/function.